# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/tests/browser/e10s/browser_obj_group.js
# Commit: c0a7fdc4bfca
# Full Hash: c0a7fdc4bfca0596ffecd347f0e248440e371989
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2022-01-05 09:33:53
# Regressor Bug: 1737944
# File Overlap Count: 2
# Description:
#   Bug 1737944 - P5: Unify GetLevelInternal and GetDefaultLevel as Accessible::GetLevel. r=Jamie
#   
#   The nsAccUtils method was a "fast" one for calculating set sizes and
#   conceptual parents. Unified it with
#   LocalAccessible::GetLevelInternal in Accessible::GetLevel with an
# ==============================================================================

diff -r f6ce0e05c92e -r c0a7fdc4bfca accessible/tests/browser/e10s/browser_obj_group.js
--- a/accessible/tests/browser/e10s/browser_obj_group.js	Tue Jan 04 21:01:35 2022 +0000
+++ b/accessible/tests/browser/e10s/browser_obj_group.js	Tue Jan 04 21:01:36 2022 +0000
@@ -21,6 +21,15 @@
   <select size="4">
     <option id="opt1">option1</option>
     <option id="opt2">option2</option>
+  </select>
+
+  <select size="4">
+    <optgroup id="select2_optgroup" label="group">
+      <option id="select2_opt1">option1</option>
+      <option id="select2_opt2">option2</option>
+    </optgroup>
+    <option id="select2_opt3">option3</option>
+    <option id="select2_opt4">option4</option>
   </select>`,
   async function(browser, accDoc) {
     let getAcc = id => findAccessibleChildByID(accDoc, id);
@@ -39,6 +48,13 @@
     // HTML select
     testGroupAttrs(getAcc("opt1"), 1, 2);
     testGroupAttrs(getAcc("opt2"), 2, 2);
+
+    // ////////////////////////////////////////////////////////////////////////
+    // HTML select with optgroup
+    testGroupAttrs(getAcc("select2_opt3"), 1, 2, 1);
+    testGroupAttrs(getAcc("select2_opt4"), 2, 2, 1);
+    testGroupAttrs(getAcc("select2_opt1"), 1, 2, 2);
+    testGroupAttrs(getAcc("select2_opt2"), 2, 2, 2);
   }
 );
 